tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -662,7 +662,7 @@ def prefetch(self, buffer_size):
     return PrefetchDataset(self, buffer_size)
 
   @staticmethod
-  def list_files(file_pattern, shuffle=None, seed=None):
+  def list_files(file_pattern, shuffle=None, seed=None, buffer_size=8):
```

Comment:
Unfortunately this would be a subtle breaking change for existing users. In the current version, the filenames are shuffled uniformly (`buffer_size` is equal to the number of filenames) whereas this would use a window of size 8. For programs that depend on shuffling the input files to get sufficient randomness, switching to a small default buffer size might harm convergence.

Reply:
@mrry Thanks for your review! Yes, the uniformly shuffling is required by some cases and may be important/necessary for model training. In this PR, it looks impossible to uniformly shuffle the filenames generated by `MatchingFilesDataset` Op as we can not know the number of filenames in advance until reaching the end of `MatchingFilesDataset` iterator. Consider these, the current version may be better. Do you have any suggestion?